colexec: TestMergeJoinerAgainstProcessor failed

Saw a reproducible on current master failure:
```
=== RUN   TestMergeJoinerAgainstProcessor
--- join type = FULL_OUTER onExpr = "" filter = "@5 = 'e'N[a\\rNaN\\x10''" seed = 4829897133613591500 run = 2 ---
CREATE TABLE left(a integer, b "char", c "char");
CREATE TABLE right(a integer, b "char", c "char");
INSERT INTO left VALUES
(1103164611, 'y#', e' co\x0f>MT'),
(-1565006090, e'rC3!a\x05\x1e\fi', '(]KH`L'),
(1204663163, e'lN\x1c@~"', '%3`'),
(1468221204, e'j1\x17\x17', e'\x06\x14'),
(886130321, e'\\Nn%', e'\x12U\x03C'),
(NULL, 'W', e'j_,\x10w'),
(1006345766, e'19\x06\x7f\x07\x10y\x1e', e'\x1fgdpO3\x1aa6'),
(1, '/', ' 7'),
(1994356194, e'#x.\x11M[T4', e'\x01%n\x04'),
(-2004744569, e'\x0fl\x10\x19x,l', e'A\x11Clh\x1f');
INSERT INTO right VALUES
(2147483647, e'v\x0eA', e'\x1cU'),
(0, 'l@tf,', NULL),
(738516335, e'h\x04\x12u\\\x02(', 'a'),
(-1498981068, 'HJ', '<C'),
(467334683, e'\x1as]N\x1fC8', '@,bOb(&x'),
(-2147483648, e'\x1aRJ;\x11-`rP', e'\x02'),
(-882647303, e'\t', e'477\x04'),
(-1127096114, NULL, NULL),
(-1456007196, NULL, e'7\t<4\fe'),
(-883064187, NULL, '');
--- FAIL: TestMergeJoinerAgainstProcessor (6.00s)
    columnar_operators_test.go:535: at or near "n": syntax error
```
Not immediately sure where the error is coming from.

--- a/pkg/sql/distsql/columnar_operators_test.go
+++ b/pkg/sql/distsql/columnar_operators_test.go
@@ -606,9 +606,13 @@ func generateFilterExpr(
 		}
 		constDatum := sqlbase.RandDatum(rng, &colTypes[colIdx], true /* nullOk */)
 		constDatumString := constDatum.String()
-		if strings.Contains(constDatumString, "NaN") || strings.Contains(constDatumString, "Inf") {
-			// We need to surround special values with quotes.
-			constDatumString = fmt.Sprintf("'%s'", constDatumString)
+		switch colTypes[colIdx].Family() {
+		case types.FloatFamily, types.DecimalFamily:
+			if strings.Contains(strings.ToLower(constDatumString), "nan") ||
+				strings.Contains(strings.ToLower(constDatumString), "inf") {
+				// We need to surround special numerical values with quotes.
+				constDatumString = fmt.Sprintf("'%s'", constDatumString)
+			}
 		}
 		return execinfrapb.Expression{Expr: fmt.Sprintf("@%d %s %s", colIdx+1, comparison, constDatumString)}
 	}

